sklearn/feature_selection/_univariate_selection.py

```diff
@@ -297,15 +297,30 @@ def f_regression(X, y, *, center=True):
     else:
         X_norms = row_norms(X.T)
 
+    if np.any(X_norms == 0) or np.ptp(y) == 0:
+        # The Pearson correlation coefficient is undefined for random
+        # variables (attributes and target) with a standard deviation
+        # of zero
+        raise ValueError("The correlation is undefined for features "
+                         "or target with identically constant values.")
+
     # compute the correlation
     corr = safe_sparse_dot(y, X)
     corr /= X_norms
     corr /= np.linalg.norm(y)
 
-    # convert to p-value
     degrees_of_freedom = y.size - (2 if center else 1)
-    F = corr ** 2 / (1 - corr ** 2) * degrees_of_freedom
-    pv = stats.f.sf(F, 1, degrees_of_freedom)
+
+    # The degrees of freedom must be at least one for the F-test
+    if degrees_of_freedom < 1:
+        raise ValueError("The sample size must be greater than two if "
+                         f"'center=True'. Got n_samples={n_samples}.")
+
+    # Convert to F-score and then to p-value
+    with np.errstate(divide="ignore"):
+        F = corr ** 2 / (1 - corr ** 2) * degrees_of_freedom
```

Comment:
We should probably have a test to check that we don't raise anymore a warning then?

Reply:
Indeed. I have included a test to ensure that `RuntimeWarning: invalid value encountered in true divide` does not get raised for `corr` equal to 1.

Thanks @glemaitre 😉.